src/io/flutter/run/MainFile.java

```diff
@@ -120,16 +121,22 @@ public boolean hasFlutterImports() {
 
   @Nullable
   private static VirtualFile findAppDir(@Nullable VirtualFile file, @NotNull Project project) {
+    if (WorkspaceCache.getInstance(project).isBazel()) {
+      Workspace workspace = WorkspaceCache.getInstance(project).get();
+      if (workspace != null) {
+        return workspace.getRoot();
+      }
+    }
+
     for (VirtualFile candidate = file; inProject(candidate, project); candidate = candidate.getParent()) {
-      if (isAppDir(candidate, project)) return candidate;
+      if (isAppDir(candidate)) return candidate;
     }
     return null;
   }
 
-  private static boolean isAppDir(@NotNull VirtualFile dir, @NotNull Project project) {
+  private static boolean isAppDir(@NotNull VirtualFile dir) {
```

Comment:
comment that this method is only appropriate for non-bazel projects or continue to require passing in the project and asserting. I'd hate for someone else to start using this method in a place where it is triggered for a bazel project.